framework/update/helper/platform_mac.cpp

```
@@ -0,0 +1,87 @@
+    int rc = posix_spawn(&pid, path, nullptr, nullptr, argv, environ);
+    if (rc != 0) {
+        return -1;
+    }
+    int status = 0;
+    waitpid(pid, &status, 0);
+    return WIFEXITED(status) ? WEXITSTATUS(status) : -1;
+}
+}
+
+namespace platform {
+bool waitForProcessExit(long long pid, int timeoutMs)
```

Cppcheck (2.21.0)

[style] The function 'waitForProcessExit' is never used.

(unusedFunction)